### PR TITLE
feat: add .properties file format plugin

### PR DESCRIPTION
**Pull request by nbouvrette**
## Summary

Adds `@inlang/plugin-properties-file` — a plugin that enables using Java `.properties` files for storing translations in inlang/Paraglide projects.

- Built on the [`properties-file`](https://github.com/properties-file/properties-file) npm package for robust parsing
- Variable interpolation with `{variable}` syntax (matching inlang message format conventions)
- Proper escaping of control characters (`\n`, `\r`, `\t`, `\\`) in exported values
- Explicit error on multi-variant messages (plural/select not supported by `.properties` format)
- Optional key sorting (ascending/descending)
- 21 tests covering roundtrip integrity, multiple locales, special characters, unicode, escaping, and edge cases

### Why .properties?

The `.properties` format is universally supported by every major TMS (Crowdin, Phrase, Lokalise, Transifex) — avoiding format lock-in. It also provides **inline comments for translator context**, a feature unavailable in JSON-based formats:

```properties
# This heading appears above the fold — keep it under 60 characters
hero.title = The End of Coding
# Shown below the title, can be longer
hero.subtitle = Building AI-Native Organizations When Humans No Longer Write Code
```

### Configuration

```json
{
  "modules": ["@inlang/plugin-properties-file"],
  "plugin.inlang.propertiesFile": {
    "pathPattern": "./messages/{locale}.properties"
  }
}
```

### CI Note

The `@inlang/website-v2:lint` failure is a pre-existing issue on the `main` branch (unused imports/declarations from the recent landing page rewrite). A separate fix has been submitted in #4346.

### Future work (not in this PR)

- **Co-located file support**: Component-scoped `.properties` files (e.g., `./src/components/Hero/{locale}.properties`) via glob-based path patterns. This could eventually become a shared utility across format plugins.
- **Comment round-trip preservation**: Currently comments are preserved in source files but not through SDK-driven rewrites (e.g., Fink editor). Would benefit from a metadata field on the Bundle/Message types.

## Test plan

- [x] 21 vitest tests passing (roundtrip, variables, sorting, special characters, unicode, escaping, multi-variant rejection, edge cases)
- [x] `tsc --noEmit` passes
- [x] `pnpm run build` produces dist/index.js
- [x] Plugin follows existing plugin conventions (package.json scripts, esbuild config, TypeBox settings schema, marketplace manifest)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by nbouvrette** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/plugins/properties-file/src/import-export/exportFiles.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+		if (sortDirection === "asc") {
+			entries = entries.sort((a, b) => a.key.localeCompare(b.key));
+		} else if (sortDirection === "desc") {
+			entries = entries.sort((a, b) => b.key.localeCompare(a.key));
+		}
+
+		const lines: string[] = [];
+		for (const entry of entries) {
+			lines.push(`${entry.key} = ${entry.value}`);
```
[the remark this answers is not in the document]
Comment (nbouvrette) [reply packages/plugins/properties-file/src/import-export/exportFiles.ts:72]: Fixed in a295a33 — exported values now escape backslashes, newlines, carriage returns, and tabs via a dedicated `escapePropertyValue()` function. Added a test covering control character escaping through roundtrip.

File: packages/plugins/properties-file/src/import-export/exportFiles.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+			continue;
+		}
+
+		const variantsOfMessage = variants.filter(
+			(v) => v.messageId === message.id
+		);
+
+		// Properties files only support single variants (no selectors)
+		const variant = variantsOfMessage[0];
+		if (!variant) {
```
[the remark this answers is not in the document]
Comment (nbouvrette) [reply packages/plugins/properties-file/src/import-export/exportFiles.ts]: Fixed in a295a33 — the exporter now throws an explicit error when a message has multiple variants that can't be represented in the `.properties` format.

To clarify: the `.properties` format **can** support plurals via ICU MessageFormat syntax within values (e.g., `{count, plural, one {1 item} other {{count} items}}`). In that model, plural forms live inside a single value string — every locale has the same set of keys, which avoids the key-mismatch problem that arises when plural forms are split across separate keys.

This constraint is not specific to `.properties` — any flat key-value format (including plain JSON) has the same limitation. The existing `@inlang/plugin-json` iterates all variants and writes them to the same key, meaning the last variant silently overwrites the others. We chose to throw an explicit error instead, which is the safer behavior — failing loudly rather than silently losing translation data.

This initial implementation treats values as opaque strings with `{variable}` interpolation. ICU MessageFormat parsing could be added as a follow-up. The error is a safety net for cases where the SDK hands the exporter multi-variant data from another plugin (e.g., `plugin-icu1`).

**Comment by samuelstroschein** [comment]
@nbouvrette thanks for the PR! 

Are you willing to host the plugin in your own repo and I'll list it on inlang.com? I fear that i might need to maintain the Java properties plugin

**Comment by nbouvrette** [comment]
> @nbouvrette thanks for the PR!
> 
> Are you willing to host the plugin in your own repo and I'll list it on inlang.com? I fear that i might need to maintain the Java properties plugin

I didn't really think about this - I'm just starting to use `inlang` but as you probably noticed by the `properties-file` repo I maintain, I prefer this file type when doing localization - I think it would be a good built-in plugin to support and with AI tools the maintenance should be very low in my opinion. If you feel strongly about this I can create a separate repo for this but given I am just starting with `inlang` there might be a risk there as well for me to maintain this on the long run

**Comment by samuelstroschein** [comment]
@nbouvrette i'd prefer a separate repo. could you do that?

**Comment by nbouvrette** [comment]
@samuelstroschein turns out I put my localization project on hold for now. Should I close this PR? not sure when I will get back to it - I do think it would be useful for you to support .properties file on the other hand

**Comment by samuelstroschein** [comment]
Closing for now then. Nothing against someone hosting the properties plugin themselves. I dont want to maintain it
